pkg/kn/commands/service/configuration_edit_flags.go

```diff
@@ -67,6 +67,25 @@ type ResourceFlags struct {
 	Memory string
 }
 
+// -- singletonString Value
+// Custom implementation of flag.Value interface to prevent multiple value assignment.
+// Useful to enforce single use of flag, e.g. --image.
+type singletonString string
+
+func (s *singletonString) Set(val string) error {
+	if len(*s) > 0 {
+		return errors.New("value is already set")
```

Comment:
I'm not sure how this "feels" like on the CLI. If this is the only error printed, then it might be confusing (especially when you have a large list of options). It would be better to actually print *which* argument was duplicated (i.e. `--image` here). I think you could implement that by using a type like

```golang
type uniqueStringArg struct {  
   // which kind of argument it is
    kind string 

   // Argument value
   value string
}

func newUniqueStringArg(kind string) uniqueStringArg {
     return uniqueStringArg{kind: kind}
}
```

and then partially pre-initialize the ConfiguarionEditFlags like in 

```golang
ceF = ConfigurationEditFlags{
    Image: newUniqueStringArg("image")
}
```

and then finally adapt the error message to include the `kind`. 

Does this make sense ?

Reply:
Actually, the error already contains the flag.
```
```

Reply:
Maybe a bit more descriptive error message?

Reply:
Ah, if this is the case, then I'm fine with having the current solution.

However for the error message I woudl use something like "flag can be provided only once" (so from the POV of the user, not from the flag's POV)

Reply:
OK, anyway I do like your suggested name `uniqueStringArg` a bit more. I'll change the name, error message and finally add some unit tests.

Reply:
Reflected review feedback in https://github.com/knative/client/pull/647/commits/c68fe1dbf790fab2b341d56de8f543ad9c80378d and added unit tests in https://github.com/knative/client/pull/647/commits/c9902886ceed29d5fe57068f603dedb8500636d2.